app/components/tic-tac-toe/game/component.js

```diff
@@ -0,0 +1,137 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { cloneDeep, random } from 'lodash';
+
+export default class TicTacToeComponent extends Component {
+  @tracked blocks = Array.from({ length: 9 }, (_, id) => ({
+    symbol: null,
+    id,
+  }));
+  @tracked nextMove = '𝗫';
+  @tracked isStarted = false;
+  @tracked winner = null;
+  @tracked isSingleplayer = true;
+  @tracked computerPossibleBlocks = [];
+
+  matrix = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+
+  get shouldBeDisabled() {
+    return Boolean(this.isStarted);
+  }
+
+  get gameMode() {
+    return this.isSingleplayer ? 'Singleplayer' : 'Multiplayer';
+  }
+
+  get shouldBeAbleToClick() {
+    return Boolean(!this.winner);
+  }
+
+  get nobodyWin() {
+    return Boolean(this.blocks.find(({ id }) => id !== null));
+  }
+
+  @action
+  onChangeGameMode() {
+    this.isSingleplayer = !this.isSingleplayer;
+  }
+
+  @action
+  onClick({ target }) {
+    if (this.isSingleplayer) {
+      if (this.shouldBeAbleToClick) {
```

Comment:
For reduce conditions nesting, consider to pull up this `if` on the begin this function:
```js
if (!shouldBeAbleToClick) {
  return;
}
```